### PR TITLE
Update Integration and Build Jenkinsfile to add the publish library.

### PR DESCRIPTION
**Pull request by prudhvigodithi**
### Description
Update Integration and Build Jenkinsfile to add the publish library.
Library PR: https://github.com/opensearch-project/opensearch-build-libraries/pull/459

### Issues Resolved
Part of https://github.com/opensearch-project/opensearch-metrics/issues/56 and https://github.com/opensearch-project/opensearch-metrics/issues/51

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by prudhvigodithi** [comment]
The 6.7.0 version is now released https://github.com/opensearch-project/opensearch-build-libraries/releases/tag/6.7.0.

File: jenkins/opensearch/integ-test.jenkinsfile [hunk 1]
```diff
@@ -309,6 +314,22 @@ pipeline {
                                     echo("Download report manifest from https://ci.opensearch.org/ci/dbc/integ-test/${env.version}/${env.buildId}/${env.platform}/${env.architecture}/${env.distribution}/test-results/${env.BUILD_NUMBER}/integ-test/test-report.yml")
                                     archiveArtifacts artifacts: 'test-report.yml'
                                 }
+                                docker.withRegistry('https://public.ecr.aws/') {
+                                    docker.image(docker_images["tar"]).inside(docker_args["tar"]) {
```
[the remark this answers is not in the document]
Comment (prudhvigodithi) [reply jenkins/opensearch/integ-test.jenkinsfile:318]: Hey Peter I have done this for `distribution-build`. Unlike the `jenkins/opensearch-dashboards/distribution-build.jenkinsfile` (`AGENT_LINUX_X64 = 'Jenkins-Agent-AL2023-X64-C54xlarge-Docker-Host'`), the `jenkins/opensearch/integ-test.jenkinsfile` the agent label is derived as `env.AGENT_LABEL = agent_nodes["${env.platform}_${architecture}"]`. So either I need to add an extra `AGENT_LINUX_X64` for this or go with the flow to infer as ` agent_nodes["${env.platform}_${architecture}"]`.

File: tests/jenkins/TestOpenSearchDashboardsIntegTest.groovy [hunk 2]
```diff
@@ -26,11 +26,11 @@ class TestOpenSearchDashboardsIntegTest extends BuildPipelineTest {
 
         helper.registerSharedLibrary(
             library().name('jenkins')
-                .defaultVersion('6.5.0')
+                .defaultVersion('6.7.0')
                 .allowOverride(true)
                 .implicit(true)
                 .targetPath('vars')
-                .retriever(gitSource('https://github.com/opensearch-project/opensearch-build-libraries.git'))
+                .retriever(gitSource('https://github.com/opensearch-project//opensearch-build-libraries.git'))
```
[the remark this answers is not in the document]
Comment (prudhvigodithi) [reply tests/jenkins/TestOpenSearchDashboardsIntegTest.groovy]: Typo, fixed it in my latest commit.

File: jenkins/opensearch/integ-test.jenkinsfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (prudhvigodithi) [reply jenkins/opensearch/integ-test.jenkinsfile:318]: Also what I noticed (could be wrong) adding a seperate section with `AGENT_LINUX_X64`, will not have the manifest created. by `createUploadTestReportManifest` in the same path as its executed on a different node, again we need to stash the file for the new agent `AGENT_LINUX_X64` workspace.

[the remark this answers is not in the document]
Comment (peterzhuamazon) [reply jenkins/opensearch/integ-test.jenkinsfile:318]: I think you can just use hardcoded:
`agent_nodes["linux_x64"]`?

**Review by peterzhuamazon — APPROVED** [review]

File: jenkins/opensearch/integ-test.jenkinsfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (prudhvigodithi) [reply jenkins/opensearch/integ-test.jenkinsfile]: Thanks Peter, added the stash logic to stash the test report file and run the publish library on `agent_nodes["linux_x64"]`

**Pull request merged**